codec-http2/src/main/java/io/netty/handler/codec/http2/HttpUtil.java

```diff
@@ -324,6 +326,32 @@ public boolean visit(Entry<CharSequence, CharSequence> entry) throws Exception {
     }
 
     /**
+     * Converts HTTP/1.1 trailing headers into HTTP/2 headers
+     */
+    public static Http2Headers toHttp2Headers(LastHttpContent in) throws Exception {
+        final Http2Headers out = new DefaultHttp2Headers();
+        final HttpHeaders inHeaders = in.trailingHeaders();
+
+        inHeaders.forEachEntry(new EntryVisitor() {
+            @Override
+            public boolean visit(Entry<CharSequence, CharSequence> entry) throws Exception {
+                final AsciiString aName = AsciiString.of(entry.getKey()).toLowerCase();
```

Comment:
I'm not sure this is necessary. It is converted to an `AsciiString` which implies a certain encoding already. WDYT?